### PR TITLE
M2_2097 [Mobile] Implement user events tracking

### PR DESCRIPTION
**Pull request by BamMironov**

### CODE REVIEW

File: src/features/pass-survey/model/hooks/useUserActionManager.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  function updateUserActionsWithAdditionalAnswer(step: number, answer: string) {
+    const actions = activityState!.actions;
+    const currentPipelineItem = activityState!.items[step];
+    const lastUserAction = actions[actions.length - 1] as
+      | SetAnswerAction
+      | undefined;
+
+    const shouldUpdateLastAction =
+      lastUserAction &&
+      lastUserAction?.payload?.activityItemId === currentPipelineItem.id;
```
Comment (anq83) [inline src/features/pass-survey/model/hooks/useUserActionManager.ts:72]: Should this check be here like it's above?
ITEMS_TO_UPDATE_USER_ACTION.includes(...

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/entities/activity/api/useActivityAnswersMutation.ts [hunk 2]
```diff
@@ -35,16 +38,11 @@ const isFileUrl = (value: string) => {
...
-    const mediaAnswer = answer?.value as MediaValue;
-    const mediaFilter = answerFilter?.value as MediaValue;
-    mediaAnswer.uri !== mediaFilter.uri;
-  });
+): AnswerDto[] => answers.filter(answer => answer !== answerFilter);
 
 const uploadAnswerMediaFiles = async (body: SendAnswersInput) => {
   for (const itemAnswer of body.answers) {
-    if (!itemAnswer) {
+    if (!isPlainObject(itemAnswer)) {
```
[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/activity/api/useActivityAnswersMutation.ts:45]: This was done in order to give Typescript a hint that itemAnswer should be of object type in the further lines of code. But `if (!itemAnswer) {` will be working anyway.

File: src/entities/activity/api/useActivityAnswersMutation.ts [hunk 3]
```diff
@@ -35,16 +38,11 @@ const isFileUrl = (value: string) => {
...
       continue;
     }
```
[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/activity/api/useActivityAnswersMutation.ts:47]: Will fix it

File: src/features/pass-survey/model/hooks/useUserActionManager.ts (hunk 1 above)
Comment (BamMironov) [reply src/features/pass-survey/model/hooks/useUserActionManager.ts:72]: No

File: src/entities/activity/api/useActivityAnswersMutation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (anq83) [reply src/entities/activity/api/useActivityAnswersMutation.ts:45]: I'd prefer 
if (!itemAnswer)

[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/activity/api/useActivityAnswersMutation.ts:45]: I see but we will get this error in this case
image

that is why we need to exp tell Typescript that the thing is not a primitive

**Pull request merged**